contrib/inventory/foreman.py

```diff
@@ -271,6 +290,100 @@ def to_safe(self, word):
 
     def update_cache(self, scan_only_new_hosts=False):
         """Make calls to foreman and save the output in a cache"""
+        use_inventory_report = self._use_inventory_report()
+        if use_inventory_report:
+            self._update_cache_inventory(scan_only_new_hosts)
+        else:
+            self._update_cache_host_api(scan_only_new_hosts)
+
+    def _update_cache_inventory(self, scan_only_new_hosts):
+        self.groups = dict()
+        self.hosts = dict()
+        host_data = self._post_request()
+
+        for host in host_data:
+            if host['Name'] in self.cache.keys() and scan_only_new_hosts:
+                continue
+            dns_name = host['Name']
+
+            host_params = host.get('all_parameters', {})
+            fact_data = host.get('Facts', {})
+            content_facet_data = host.get('content_facet_attributes', {})
+            content_facet_attributes = {}
+            for cf in content_facet_data:
+                content_facet_attributes.update({cf[0]: cf[1]})
+
+            fact_list = {}
+            for fact in fact_data:
+                fact_list.update({fact[0]: fact[1]})
+
+            # Create ansible groups for hostgroup
+            group = 'hostgroup'
+            val = host.get('%s_title' % group) or host.get('%s_name' % group)
+            if val:
+                safe_key = self.to_safe('%s%s_%s' % (
+                    to_text(self.group_prefix),
+                    group,
+                    to_text(val).lower()
+                ))
+                self.inventory[safe_key].append(dns_name)
+
+            # Create ansible groups for environment, location and organization
+            for group in ['environment', 'Location', 'Organization']:
```

Comment:
I find the inconsistent capitalization very confusing. There's `Name` and `Facts` but also `all_parameters`. Here we have `environment` but also `Location` and `Organization`. Ideally the original API response would be normalized.